File: src/core/interfaces/types/props.ts
import { Position, Viewport } from './common';
import { TileState, POIData } from './state';

export interface MapContainerProps {
  className?: string;
}

export interface ViewportManagerProps {
  viewport: Viewport;
  onViewportChange: (viewport: Viewport) => void;
  children: React.ReactNode;
  className?: string;
}

export interface TileLayerProps {
  tiles: Record<string, TileState>;
  viewport: Viewport;
  selectedTile: Position | null;
  hoveredTile: Position | null;
  onTileClick?: (position: Position) => void;
  onTileHover?: (position: Position) => void;
  className?: string;
}

export interface TileProps {
  state: TileState;
  selected: boolean;
  hovered: boolean;
  onClick?: () => void;
  onHover?: () => void;
  className?: string;
}

export interface POILayerProps {
  pois: Record<string, POIData>;
  viewport: Viewport;
  onPOIClick?: (id: string) => void;
  onPOIHover?: (id: string | null) => void;
  className?: string;
}

export interface FogOfWarProps {
  tiles: Record<string, TileState>;
  viewport: Viewport;
  currentPosition: Position;
  visionRadius: number;
  className?: string;
}

File: backend/app/src/core/interfaces/types/props.py
from typing import Any, Dict, Union


class MapContainerProps:
    className?: str
class ViewportManagerProps:
    viewport: Viewport
    onViewportChange: (viewport: Viewport) => None
    children: React.ReactNode
    className?: str
class TileLayerProps:
    tiles: Dict[str, TileState>
    viewport: Viewport
    selectedTile: Union[Position, None]
    hoveredTile: Union[Position, None]
    onTileClick?: (position: Position) => None
    onTileHover?: (position: Position) => None
    className?: str
class TileProps:
    state: TileState
    selected: bool
    hovered: bool
    onClick?: () => None
    onHover?: () => None
    className?: str
class POILayerProps:
    pois: Dict[str, POIData>
    viewport: Viewport
    onPOIClick?: (id: str) => None
    onPOIHover?: Union[(id: str, None) => None]
    className?: str
class FogOfWarProps:
    tiles: Dict[str, TileState>
    viewport: Viewport
    currentPosition: Position
    visionRadius: float
    className?: str